chrome/content/zotero/error/errorReport.jsx

```diff
@@ -0,0 +1,182 @@
+
+	const handleSubmit = useCallback(async () => {
+		setCanAdvance(false);
+		setCanRewind(false);
+		setCanCancel(false);
+		
+		const parts = {
+			error: "true",
+			errorData: errorData.join('\n'),
+			extraData: extraData,
+			diagnostic: diagnosticInfo
+		};
+		
+		let body = '';
+		for (let key in parts) {
+			body += key + '=' + encodeURIComponent(parts[key]) + '&';
+		}
+		body = body.substr(0, body.length - 1);
+		const xmlhttp = await Zotero.HTTP.request(
+			"POST",
+			ZOTERO_CONFIG.REPOSITORY_URL + "report",
+			{
+				body,
+				successCodes: false,
+				foreground: true
+			}
+		);
+
+		if (!xmlhttp.responseXML) {
+			try {
+				if (xmlhttp.status > 1000) {
+					alert(Zotero.getString('errorReport.noNetworkConnection'));
+				}
+				else {
+					alert(Zotero.getString('errorReport.invalidResponseRepository'));
+				}
+			}
+			catch (e) {
+				alert(Zotero.getString('errorReport.repoCannotBeContacted'));
+			}
+			
+			wizardRef.current.rewind();
+			return;
+		}
+		
+		var reported = xmlhttp.responseXML.getElementsByTagName('reported');
+		if (reported.length != 1) {
+			alert(Zotero.getString('errorReport.invalidResponseRepository'));
+			wizardRef.current.rewind();
+			return;
+		}
+
+		setReportID(reported[0].getAttribute('reportID'));
+		wizardRef.current.goTo('error-summary');
+	}, [errorData, extraData, diagnosticInfo]);
+
+	useEffect(() => {
+		(async () => {
+			const newDiagnosticInfo = await Zotero.getSystemInfo();
+			setDiagnosticInfo(newDiagnosticInfo);
+			const errorDataText = errorData.length
+				? data.errorData.join('\n\n')
+				: Zotero.getString('errorReport.noErrorsLogged', Zotero.appName);
+			
+			const newLogText = errorDataText + '\n\n'
+				+ (extraData !== '' ? extraData + '\n\n' : '')
+				+ newDiagnosticInfo;
+			setLogText(newLogText);
+			setCanAdvance(true);
+		})();
+	}, [data, errorData, extraData]);
+
+	return (
+		<Wizard
+			canAdvance={ canAdvance }
+			canRewind={ canRewind }
+			canCancel={ canCancel }
+			className="error-report"
+			onClose={ handleClose }
+			ref={ wizardRef }
+		>
+			<WizardPage pageId="error-intro">
+				<p className="description">{ msg }</p>
+				<textarea readOnly={ true } value={ logText }></textarea>
+				<p className="description advance-message">
+					{ Zotero.getString('errorReport.advanceMessage', nextLabel) }
+				</p>
+			</WizardPage>
+			<WizardPage pageId="error-progress" onPageShow={ handleSubmit }>
+				<p className="description">
+					{ Zotero.getString('errorReport.submissionInProgress') }
+				</p>
+			</WizardPage>
+			<WizardPage pageId="error-summary">
+				<p className="description">
+					{ Zotero.getString('errorReport.submitted') }
+				</p>
+				<div className="report-result">
+					<label htmlFor={ id.current + '-report-result' }>
+						{ Zotero.getString('errorReport.reportID') }
+					</label>
+					<input
+						className="report-id"
+						id={ id + '-report-result' }
+						readOnly={ true }
+						value={ reportID }
+					/>
```

Comment:
Why not just wrap the `input` in the `label` and skip the `for` (here and elsewhere)?

Reply:
Both are perfectly valid, however few reasons to keep explicit `for`:

* Recommended by [W3C for accessibility reasons](https://www.w3.org/WAI/tutorials/forms/labels/#associating-labels-implicitly)
* Generally easier to style this way
* We're [using](https://github.com/zotero/web-library/issues/134#issuecomment-481601752) explicit `for` in Web Library 

I actually came to prefer it this way, not only for the above reasons, but also because, semantically, it makes little sense for an input to be part of a label that describes it (other way round would actually make more sense!)

I'd recommend keeping explicit `for` but can change it if it's a problem.

Reply:
Yeah, I've seen the better-supported-by-accessibility-tools argument. That seems like a problem with the tools, not the method, and hopefully one that's been fixed in the years since all the mentions of those limitations. It's just particularly annoying in the context of React because of the need for id mechanism, but maybe worth it for the easier styling. (And the semantic argument is fair! XUL did get this right…)